subsys/bluetooth/controller/ll_sw/ull_adv_iso.c

```diff
@@ -1155,19 +1173,51 @@ static uint8_t ptc_calc(const struct lll_adv_iso *lll, uint32_t event_spacing,
 			uint32_t event_spacing_max)
 {
 	if (event_spacing < event_spacing_max) {
-		uint8_t ptc;
+		uint32_t ptc;
+		uint8_t nse;
 
-		/* Possible maximum Pre-transmission Subevents per BIS */
+		/* Possible maximum Pre-transmission Subevents per BIS.
+		 * sub_interval is at least T_MSS_150 + MPT (hence a value in 8 bits or more), i.e.
+		 * the below division and the subsequent multiplication with lll->bn does not
+		 * overflow.
+		 */
 		ptc = ((event_spacing_max - event_spacing) /
 		       (lll->sub_interval * lll->bn * lll->num_bis)) *
 		      lll->bn;
 
-		/* FIXME: Here we restrict to a maximum of BN Pre-Transmission
-		 * subevents per BIS
+		/* Required NSE */
+		nse = lll->bn * lll->irc; /* 3 bits + 4 bits */
+
+		/* Requested NSE is greater than Required NSE, Pre-Transmission offset has been
+		 * provided.
+		 *
+		 * NOTE: This is the case under HCI test command use to create BIG, i.e. test_config
+		 *       variable is true.
+		 */
+		if (lll->nse > nse) {
+			/* Restrict PTC to number of available subevents */
+			ptc = MIN(ptc, lll->nse - nse);
+		} else {
+			/* No PTO requested, Zephyr Controller implementation here will try using
+			 * Pre-Transmisson offset of BT_CTLR_ADV_ISO_PTO_MIN, i.e. restrict to a
+			 * maximum of BN Pre-Transmission subevents per BIS. This allows for a
+			 * better time diversity ensuring skipped or missing reception at the ISO
+			 * Sync Receiver so it can still have another chance at receiving the ISO
+			 * PDUs within the permitted maximum transport latency.
+			 *
+			 * Usecases where BAP Broadcast Audio Assistant role device has a drifting
+			 * ACL Peripheral role active in the BAP Broadcast Audio Sink device.
+			 */
+			ptc = MIN(ptc, (lll->bn * BT_CTLR_ADV_ISO_PTO_GROUP_COUNT));
+		}
+
+		/* FIXME: Do not remember why ptc is 4 bits, it should be 5 bits as ptc is a
+		 *        running buffer offset related to nse. Fix ptc and ptc_curr definitions,
+		 *        until then lets have an assert check here.
 		 */
-		ptc = MIN(ptc, lll->bn);
+		LL_ASSERT(ptc <= BIT_MASK(4));
 
-		return ptc;
+		return ptc & BIT_MASK(5);
```

Comment:
Is the `& BIT_MASK(5)` valuable when you already have the assert?

Reply:
agree, fixed.